references/optical_flow/transforms.py

```diff
@@ -0,0 +1,267 @@
+
+        return img1, img2, flow, valid_flow_mask
+
+
+class AsymmetricColorJitter(T.ColorJitter):
+    def __init__(self, brightness=0, contrast=0, saturation=0, hue=0, p=0.2):
+        super().__init__(brightness=brightness, contrast=contrast, saturation=saturation, hue=hue)
+        self.p = p
+
+    def forward(self, img1, img2, flow, valid_flow_mask):
+
+        if torch.rand(1) < self.p:
+            # asymmetric: different transform for img1 and img2
+            img1 = super().forward(img1)
+            img2 = super().forward(img2)
+        else:
+            # symmetric: same transform for img1 and img2
+            batch = torch.stack([img1, img2])
+            batch = super().forward(batch)
+            img1, img2 = batch[0], batch[1]
+
+        return img1, img2, flow, valid_flow_mask
+
+
+class RandomErase(torch.nn.Module):
+    def forward(self, img1, img2, flow, valid_flow_mask):
+        bounds = [50, 100]
+        ht, wd = img2.shape[:2]
+
+        # Warning : This won't work with image values in [0, 1] because of round()
+        mean_color = img2.view(3, -1).float().mean(axis=-1).round()
+        for _ in range(torch.randint(1, 3, size=(1,)).item()):
+            x0 = torch.randint(0, wd, size=(1,)).item()
+            y0 = torch.randint(0, ht, size=(1,)).item()
+            dx, dy = torch.randint(bounds[0], bounds[1], size=(2,))
+            img2[:, y0 : y0 + dy, x0 : x0 + dx] = mean_color[:, None, None]
+
+        return img1, img2, flow, valid_flow_mask
+
+
+class RandomHorizontalFlip(T.RandomHorizontalFlip):
+    def forward(self, img1, img2, flow, valid_flow_mask):
+        if torch.rand(1) > self.p:
+            return img1, img2, flow, valid_flow_mask
+
+        img1 = F.hflip(img1)
+        img2 = F.hflip(img2)
+        flow = F.hflip(flow) * torch.tensor([-1, 1])[:, None, None]
+        if valid_flow_mask is not None:
+            valid_flow_mask = F.hflip(valid_flow_mask)
+        return img1, img2, flow, valid_flow_mask
+
+
+class RandomVerticalFlip(T.RandomVerticalFlip):
+    def forward(self, img1, img2, flow, valid_flow_mask):
+        if torch.rand(1) > self.p:
+            return img1, img2, flow, valid_flow_mask
+
+        img1 = F.vflip(img1)
+        img2 = F.vflip(img2)
+        flow = F.vflip(flow) * torch.tensor([1, -1])[:, None, None]
+        if valid_flow_mask is not None:
+            valid_flow_mask = F.vflip(valid_flow_mask)
+        return img1, img2, flow, valid_flow_mask
+
+
+class MaybeResizeAndCrop(torch.nn.Module):
+    # This transform will resize the input with a given proba, and then crop it.
+    # These are the reversed operations of the built-in RandomResizedCrop,
+    # although the order of the operations doesn't matter too much.
+    # The reason we don't rely on RandomResizedCrop is because of a significant
+    # difference in the parametrization of both transforms.
+    #
+    # There *is* a mapping between the inputs of MaybeResizeAndCrop and those of
```

Comment:
I am confused by this comment. Does this imply that there is a way to map the parameters between the two transforms (just the arguments and NOT the behaviours)? Or does this imply that there is a way to take the inputs of one transform and convert them to the input of the other so that they both behave the same (aka re-parameterization)? 

I guess you mean the former. Also why is your argument here more about different distributions rather than the fact that the order of operations is different?

Reply:
It's the former, I'll clarify.

> Also why is your argument here more about different distributions rather than the fact that the order of operations is different?

The order of operation doesn't matter much in practice: resizing a crop or cropping a resized image would give the same result, up to some artifacts at the borders of the output (due to interpolation).

Here the problem we have is really about the distribution of the randomly-sampled parameters. Here, scale_x and scale_y are randomly sampled and their density looks like this

![image](https://user-images.githubusercontent.com/1190450/144840028-87981dfc-24bb-4952-bff6-5570dc34956f.png)


Now if I map these to the current parametrization of our RandomResizedCrop, this would be equivalent to the following distribution for the `scale` parameter

![image](https://user-images.githubusercontent.com/1190450/144840133-38ef69c1-8303-4c75-8051-a427ad1955a7.png)


and the following for the `ratio` parameter

![image](https://user-images.githubusercontent.com/1190450/144840158-f65cff33-12d7-4f05-ad83-a2090439f032.png)

Unfortunately, `RandomResizedCrop` is unable to sample according to these 2 distributions, no matter what we pass as values.